diff --git a/dist/backends/store/fs.js b/dist/backends/store/fs.js
index b418cfe79836c1948518eff13e83fc6020437191..5b2db8d93e8a8db2e2191aff177e641dcf0c1abb 100644
--- a/dist/backends/store/fs.js
+++ b/dist/backends/store/fs.js
@@ -896,9 +896,11 @@ export class StoreFS extends FileSystem {
         if (this.attributes.has('no_id_tables'))
             return await this._findInode(tx, path);
         const ino = this._ids.get(path);
-        if (ino === undefined)
-            throw withErrno('ENOENT');
-        return new Inode((await tx.get(ino)) ?? _throw(withErrno('ENOENT')));
+        if (ino !== undefined)
+            return new Inode((await tx.get(ino)) ?? _throw(withErrno('ENOENT')));
+        const inode = await this._findInode(tx, path);
+        this._add(inode.ino, path);
+        return inode;
     }
     /**
      * Finds the Inode of `path`.
@@ -910,9 +912,11 @@ export class StoreFS extends FileSystem {
         if (this.attributes.has('no_id_tables'))
             return this._findInodeSync(tx, path);
         const ino = this._ids.get(path);
-        if (ino === undefined)
-            throw withErrno('ENOENT');
-        return new Inode(tx.getSync(ino) ?? _throw(withErrno('ENOENT')));
+        if (ino !== undefined)
+            return new Inode(tx.getSync(ino) ?? _throw(withErrno('ENOENT')));
+        const inode = this._findInodeSync(tx, path);
+        this._add(inode.ino, path);
+        return inode;
     }
     _lastID;
     /** Allocates a new ID and adds the ID/path */
