diff --git a/dist/replicad.d.ts b/dist/replicad.d.ts
--- a/dist/replicad.d.ts
+++ b/dist/replicad.d.ts
@@ -2217,13 +2217,13 @@ export declare class Sketches {
         extrusionProfile?: ExtrusionProfile;
         twistAngle?: number;
         origin?: Point;
-    }): AnyShape;
+    }): Shape3D;
     /**
      * Revolves the drawing on an axis (defined by its direction and an origin
      * (defaults to the sketch origin)
      */
     revolve(revolutionAxis?: Point, config?: {
         origin?: Point;
         angle?: number;
-    }): AnyShape;
+    }): Shape3D;
 }
