diff --git a/dist/replicad.d.ts b/dist/replicad.d.ts
index 6307f486c39a1a838bf71a3a0f0231a200c0e15e..fe355e743e88c5077a8e76983e22669c2d35f266 100644
--- a/dist/replicad.d.ts
+++ b/dist/replicad.d.ts
@@ -2002,14 +2002,14 @@ export declare class Sketches {
         extrusionProfile?: ExtrusionProfile;
         twistAngle?: number;
         origin?: Point;
-    }): AnyShape;
+    }): Shape3D;
     /**
      * Revolves the drawing on an axis (defined by its direction and an origin
      * (defaults to the sketch origin)
      */
     revolve(revolutionAxis?: Point, config?: {
         origin?: Point;
-    }): AnyShape;
+    }): Shape3D;
 }
 
 /**
